### ISSUE TITLE
Notification on Idle and on Movement

### ISSUE THREAD

**Issue by Abyss777** [vitalidze/traccar-web#95]
It would be nice to have notifications on this events:
- The Tracker is Idle more then some minutes (User should point the number of minutes in Settings)
- The Tracker is moving in time that it shouldn't (User should point the schedule then moving is prohibited in Settings)

Examples:
- The tracker is installed on the machine carrier. His manager wants to know then his courier isn't working but drinking coffee (more than 10 minutes). It would be nice to be able to specify the timetable for the courier.
  From 8:00 to 12:00 and from 12:30 to 17:00 from monday till friday.
- The same situation. Boss want to know if somebody has taken the car at night. (from 17:00 to 08:00)
- The tracker is installed on private car. Its owner wants to now then his car stolen. The schedule will be from 23:00 to 07:00 (car near house) and from 10:00 to 17:00 (car near work) from monday till friday.

**Comment by vitalidze** [comment]
This intersects with #25 (have no idea on how to implement it correctly yet), #60 and #88 

Right now my next goal is to implement geofences.

**Comment by Abyss777** [comment]
How frequently is updating `position.IdleSince` ? (it happends in `MapController.update()`)
Is it possible to do the work in `NotificationServiceImpl.NotificationSender` ?

**Comment by vitalidze** [comment]
It is updated with `ApplicationSettings.updateInterval` frequency (15 seconds by default). For the first time there is also a map of latest non-idle positions by device loaded from server (see `DataServiceImpl.getLatestNonIdlePositions`). This is used to calculate how much time device/vehicle is not moving (or when it stopped).

Well, it can be done in `NotificationSender`, but my approach is to split notification sending and notification events posting. So something posts events and then `NotificationSender` sends them to the users that expect to receive it (i.e. have access to the device, set up email and checked `notifications` box in his profile).

**Comment by Abyss777** [comment]
Yes, sure, my mistake. Best way is creating new class in `EventServiceImpl` like `MovingDetector` and `IdleDetector` to rise new types of Events.
- Need to expand the device settings to store timetables then events should be rised
- Need to find cool library with visual control of timetable

A lot of work :( May be after geofences.
